### ISSUE TITLE
Suggestion: Ideas for a more comprehensive MechJeb

### ISSUE THREAD

**Issue by AppleDavidJeans** [MuMech/MechJeb2#388]
I have a bunch of suggestions for enhancing and improving this plugin's wonderful abilities:

1a) Edit the Translatron's hold speed feature so that it can also account for jet engines' behavior (no more throttling up or down around the speed when on jet engines), or
1b) Add an option in the Spaceplane Guidance module allowing one to control a plane's speed, using logic that can account for the different thrust behaviors of rocket and jet engines. (I'd prefer this one)
2) Try to integrate arrowstar's work in his KSP Trajectory Optimization Tool into MechJeb so that the plugin has its own advanced planetary flyby module on par with KSP TOT. Since this could be a reeeeeeeally huge thing to add, it leads me to #3.
3) Make an API or something that would allow for "mods within a mod" to allow other people to add their own modules or modification of existing modules. This could open up a whole new realm of possibilities for MechJeb. For example, the authors of mods such as Kerbal Alarm Clock, KSPTOT, Final Frontier, Crew Manifest, etc., could make a version of his or her mod(s) designed for use in MechJeb. And also, to give everyone an opportunity to use those mods if the authors decide to develop said versions, don't allow for external (as in outside of KSP) software dependencies.
4) Back to the Spaceplane Guidance module: Enhance the auto-land feature to allow players to make their own approaches at other places like the airbase in the North polar region in the Kerbin-Side mod, or a base on Laythe.
5) In the same module, make a feature that allows players to make their own automated ascent profiles for their SSTO's.

Basically, that's about all I can think of.
